### DISCUSSION TITLE
Airflow PostgresOperator SELECT statement

### DISCUSSION THREAD

**Discussion by Reslan-Tinawi** [apache/airflow#25863] [Q&A] [answered]
Hello, Airflow community.

I was reading this article: [How-to Guide for PostgresOperator
](https://airflow.apache.org/docs/apache-airflow-providers-postgres/stable/operators/postgres_operator_howto_guide.html)

I understand that we can pass a statement to the `sql` argument, which will then be executed. However, it's not clear to me how this would work for `SELECT` statements.

My question is: where would be the result set of the `SELECT` query?

Reading the `execute` function [code](https://github.com/apache/airflow/blob/874a95cc17c3578a0d81c5e034cb6590a92ea310/airflow/providers/postgres/operators/postgres.py#L73) of `PostgresOperator`, it doesn't seem that any result is actually returned.

Right now, I'm using `PythonOperator` with `PostgresHook` and the `get_pandas_df` method, to retrieve the data, and apply different transformations.

Is this the right way to handle data transformation or are there best practices I should follow? Additionally, in what scenario I would use `SELECT` in `PostgresOperator`? To check if a table exists, to check if has rows, etc.?

Thank you 🙏🏼

**Comment by potiuk** [comment, marked as answer]
PostgresOperator is not too useful for selects - it's good for DML and DDL statements but not for DQL statements. This is not how Airlfow works because each task works in isolation and in order to send output to another task, it needs to send it via Xcom (which is also a database operation), So it makes little sense to use predefined PostgresOperator to run select query because indeed you wouldn't be able to do much with the output. 

There are a number of specific operators (called transfer operators) -  for example PostgresToGCSOperator that can  send the retrieved data somewhere - in this example, they  can produce CSV file in GCS. and if you want to do somethinig like that, maybe an operator exists for that already. But even if not, you can write your own.

I personally recommend TaskFlow approach for that. I wrote a blog post recently describing it, so you can read it here: https://medium.com/apache-airflow/generic-airflow-transfers-made-easy-5fe8e5e7d2c2

**Reply by Reslan-Tinawi** [reply]
Thanks for your answer.

I have already written my custom operator to move data from Microsoft SQL server to Postgres. 

I was just curious about the `PostgresOperator` and its usage.
